### ISSUE TITLE
False positive on raising-bad-type

### ISSUE THREAD

**Issue by MartinBonner** [PyCQA/pylint#4189]
### Steps to reproduce

sample.py:
```
"Sample program"

def returner():
    """returner"""
    if 3 > 4:
        return Exception("Unexpected")
    return None

def user():
    """user"""
    err = returner()
    if err is not None:
        raise err
```

Command:  `pylint sample.py`

### Current behavior
Actual output:
```
************* Module x.sample
sample.py:13:8: E0702: Raising NoneType while only classes or instances are allowed (raising-bad-type)
```
Pylint has seen a comparison with NoneType, but hasn't picked up that the comparison is precisely to prevent the error it is complaining about!

### Expected behavior
No output

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.8.5 (default, Jan 27 2021, 15:41:15)
[GCC 9.3.0]
```

**Comment by MartinBonner** [comment]
Duplicate of #2529
